tokio/src/runtime/queue.rs

```diff
@@ -79,6 +52,13 @@ const LOCAL_QUEUE_CAPACITY: usize = 4;
 
 const MASK: usize = LOCAL_QUEUE_CAPACITY - 1;
 
+fn make_fixed_size<T>(buffer: Box<[T]>) -> Box<[T; LOCAL_QUEUE_CAPACITY]> {
+    assert_eq!(buffer.len(), LOCAL_QUEUE_CAPACITY);
+
+    // SAFETY: We check that the length is correct.
+    unsafe { Box::from_raw(Box::into_raw(buffer).cast()) }
+}
```

Comment:
is this really necessary? is there a reason we can't just replace the `Vec::new` with a boxed array literal? if we can't do that, it would be nice to have a comment explaining why, IMO.

Reply:
This is the alternative:
```Rust
Box::new([
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
    UnsafeCell::new(MaybeUninit::new()),
])
```

Reply:
ah, um. on second thought let's maybe not do that.